go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -1010,30 +1010,64 @@ func (tsv *TabletServer) Execute(ctx context.Context, target *querypb.Target, sq
 			if err != nil {
 				return err
 			}
-			qre := &QueryExecutor{
-				query:          query,
-				marginComments: comments,
-				bindVars:       bindVariables,
-				transactionID:  transactionID,
-				options:        options,
-				plan:           plan,
-				ctx:            ctx,
-				logStats:       logStats,
-				tsv:            tsv,
-			}
-			extras := tsv.watcher.ComputeExtras(options)
-			result, err = qre.Execute()
-			if err != nil {
-				return err
+			if plan.Reason == planbuilder.ReasonTopic {
```

Comment:
I tried to hook this logic into a few different places and this was the best. The only downside is that it is looking for a single result. I'm not sure how to logically return 0 to N inserts back to the caller, so I chose to just return the last one.

Reply:
The purpose of `Reason` is display only. You should instead look at planID, which should already be set to `InsertTopic` from the way I see you're building the plan.

Reply:
👍 I just was following other patterns I saw. There were very few errors being returned, even in cases where it seemed like there should be, and Reason being set in those cases. I prefer sending back errors early.